pandas/types/missing.py

```diff
@@ -391,3 +394,36 @@ def na_value_for_dtype(dtype):
     elif is_bool_dtype(dtype):
         return False
     return np.nan
+
+
+def valid_fill_value(value, dtype, is_period=False):
```

Comment:
This just returns a `bool`, the action verb `validate` implies to me raising an error or processing the data somehow. Like in e.g. `maybe_upcast`. I can rename this, but I suggest reserving `validate_fill_value` as the name for the `fillna` sub-impl I'm moving to `core/missing.py` per your review.